### DISCUSSION TITLE
Long tap to right click triggering when it shouldn't

### DISCUSSION THREAD

**Discussion by Susko3** [ppy/osu#23460] [Q&A]
Some places where it triggers when it shouldn't:
- ~~osu! gameplay (when following slow sliders or sliders that quickly go back and forth)~~
- any button that requires long pressing (this isn't too distracting but it's weird when there are two independent progress circles)
  - `HoldForMenuButton`
  - `DangerousActionDialog`

**Comment by peppy** [comment]
I've started to investigate a solution for this, but am not sold on what I have. It will work correctly for the two cases mentioned, but potentially fail for others where `OnMouseDown` is handled (potentially just to block input from reaching something behind).

https://github.com/ppy/osu-framework/compare/master...peppy:osu-framework:fix-hold-touch-aggressiveness?expand=1

**Comment by Susko3** [comment]
- First part fixed by #23529
